GH-47157: [Docs] Improve presentation of Other available packages section in build_system.rst (#47411)
apache/arrow: docs/source/cpp/build_system.rst

There are other available packages, they can also be used with the `find_package
<https://cmake.org/cmake/help/latest/command/find_package.html>`_ directive.
This is the list of available packages:

* ArrowCUDA
* ArrowAcero
* ArrowCompute
* ArrowDataset
* ArrowFlight
* ArrowFlightSql
* ArrowFlightTesting
* ArrowSubstrait
* ArrowTesting
* Gandiva
* Parquet

> Target specific to above packages follow a consistent naming pattern:

Reviewer: ```suggestion
Usage with find_package and target names follows a consistent naming pattern:
```

* find_package usage: ``find_package(PackageName REQUIRED)``

> * shared target: ``PackageName::package_name_shared``
> * static target: ``PackageName::package_name_static``

Reviewer: ```suggestion
* Shared Target: ``PackageName::package_name_shared``
* Static Target: ``PackageName::package_name_static``
```

Reviewer: This is just to make this list consistent with the one below.

> Example, to use the ArrowFlightTesting package:
> 
> * Package Name: ``ArrowFlightTesting``
> * find_package Usage: ``find_package(ArrowFlightTesting REQUIRED)``
> * Shared Target: ``ArrowFlightTesting::arrow_flight_testing_shared``
> * Static Target: ``ArrowFlightTesting::arrow_flight_testing_static``

Reviewer: ```suggestion
For example, to use the ``ArrowCompute`` package:

* find_package Usage: ``find_package(ArrowCompute REQUIRED)``
* Shared Target: ``ArrowCompute::arrow_compute_shared``
* Static Target: ``ArrowCompute::arrow_compute_static``
```

Reviewer: An example is great, but I think a more common example like ArrowCompute would be even better.
